### DISCUSSION TITLE
NSPanel doesn't see correct blueprint

### DISCUSSION THREAD

**Discussion by piotrvvilk** [Blackymas/NSPanel_HA_Blueprint#1230] [Q&A] [answered]
Hello. I have got problem with my Nspanel. It doesn't see correct blueprint version. I succesfully uploaded tft file direct via nextion IDE, esphome device is visible on home assistant. I added blueprint and automation and nothing, I did everything 

![1](https://github.com/Blackymas/NSPanel_HA_Blueprint/assets/22833057/e1b204dc-ff0a-439e-aa6b-ee0797933a0d)
![2](https://github.com/Blackymas/NSPanel_HA_Blueprint/assets/22833057/c688b732-138c-487d-8a9c-1c91361b4149)

Regards,
Piotr

**Comment by edwardtfn** [comment]
Have you renamed your panel on Home Assistant or renamed some of it's entities?

I would suggest this:
1. Go to Settings > Devices and Services > ESPHome and then remove your panel.
2. Restart Home Assistant.
3. Go to Settings > Devices and Services > Add integration > ESPHome and add your panel's host name or ip (192.168.0.136).
4. Go to Settings > Automations, find your automation using the blueprint and select again your panel.

**Comment by piotrvvilk** [comment]
I tried :(. After another try i take info:
 
![1](https://github.com/Blackymas/NSPanel_HA_Blueprint/assets/22833057/78673169-c036-420b-8969-317c24e60f97)

There is difference in names, but it's gonna happend automatic. ESPHome change name to capital letters on beginning.

There is a difference in names, but it is automatic. ESPHome change name to capital letters on beginning.
![2](https://github.com/Blackymas/NSPanel_HA_Blueprint/assets/22833057/a18d8bb6-d071-455f-9573-cd5cac581112)
![3](https://github.com/Blackymas/NSPanel_HA_Blueprint/assets/22833057/4aa87073-d2e2-44c0-895f-edf0facfefa6)

**Reply by edwardtfn** [reply]
The problem is on the name of the services.
Just to you understand how it works, when the blueprint starts, it looks for a point with name ending with `_nextion_display` and then gets the panel name prefix from there.
Then it uses this name prefix to look for the services.
If you rename anything, this process will break.
Unfortunately, as far as I know, there's no more reliable way to get this info from Home Assistant.

With that said, could you please go to your panel page in Home Assistant (Settings > Devices and Services > ESPHome) and find the entity "Nextion display" (on Diagnostic area), get it's `entity_id` and share it here?

**Reply by piotrvvilk** [reply]
![5](https://github.com/Blackymas/NSPanel_HA_Blueprint/assets/22833057/cae4789a-68b3-433a-9502-0c9e5c1f2dcd)

**Reply by edwardtfn** [reply]
Ok, now please go to Developer Tools > Services and look for services starting by `esphome.ns_panel_`

**Reply by piotrvvilk** [reply]
What I should search for? There are many services.
![1](https://github.com/Blackymas/NSPanel_HA_Blueprint/assets/22833057/7997176e-9ed3-41d5-91fd-d4c18adc8416)

**Reply by edwardtfn** [reply]
Ok, everything looking fine.

Don't you have anything related to this at Home Assistant logs (Settings > System > Logs)?

**Reply by piotrvvilk** [reply]
I found only this:
![2](https://github.com/Blackymas/NSPanel_HA_Blueprint/assets/22833057/593c942d-4972-4c87-9979-e6eda7f00a4e)

One more thing.  I use duckdns and I have problem with hass on local IP address. It wasn't problem for me, so I never tried to fix it. Could be the reason of my problems? Nextion update url is local.
![1](https://github.com/Blackymas/NSPanel_HA_Blueprint/assets/22833057/4ce85d05-a394-4fc7-9067-c37f1c9eec49)

**Reply by piotrvvilk** [reply, marked as answer]
After deleted device, added new and another hass restart, ns-panel work great! Thanks @edwardtfn :)
